tests/functional/dispersionCoarse/dispersion1d.py

```diff
@@ -0,0 +1,379 @@
+        #    modes += v*np.cos(k*x+f)
+        #return modes
+        return 0.0
+
+    def vz(x):
+        #modes = 0.0
+        #for (k, v, f) in zip(wave_numbers, b_amplitudes, phases):
+        #    modes += v*np.sin(k*x+f)
+        #return modes
+        return 0.0
+
+    def vthx(x):
+        return 0.01
+
+    def vthy(x):
+        return 0.01
+
+    def vthz(x):
+        return 0.01
+
+
+    vvv = {
+        "vbulkx": vx, "vbulky": vy, "vbulkz": vz,
+        "vthx": vthx, "vthy": vthy, "vthz": vthz
+    }
+
+
+    MaxwellianFluidModel(
+        bx=bx, by=by, bz=bz,
+        main={"charge": 1, "density": density, **vvv, "init":{"seed": seed}}
+    )
+
+    ElectronModel(closure="isothermal", Te=0.)
+
+    sim = ph.global_vars.sim
+
+    timestamps = np.arange(0, sim.final_time+sim.time_step, 10*sim.time_step)
+
+
+    for quantity in ["E", "B"]:
+        ElectromagDiagnostics(
+            quantity=quantity,
+            write_timestamps=timestamps,
+            compute_timestamps=timestamps,
+        )
+
+
+    for quantity in ["density", "bulkVelocity"]:
+        FluidDiagnostics(
+            quantity=quantity,
+            write_timestamps=timestamps,
+            compute_timestamps=timestamps,
+            )
+
+    return wave_numbers, b_amplitudes
+
+
+
+# ___ post-processing functions
+def get_all_w(run_path, wave_numbers, polarization):
+    file = os.path.join(run_path, "EM_B.h5")
+    times = get_times_from_h5(file)
+
+    nm = len(wave_numbers)
+    print('number of modes : {}'.format(nm))
+
+    r = Run(run_path)
+    byz = np.array([])
+
+    for time in times:
+
+        interp_by, x = r.GetB(time, merged=True, interp='nearest')['By']
+        interp_bz, x = r.GetB(time, merged=True, interp='nearest')['Bz']
+        by = interp_by(x[0])
+        bz = interp_bz(x[0])
+
+        # polarization = +1 for R mode, -1 for L mode
+        byz = np.concatenate((byz, by+polarization*1j*bz))
+
+    nx = x[0].shape[0]
+    nt = times.shape[0]
+    byz = np.reshape(byz, (nt, nx))
+
+    BYZ = np.absolute(np.fft.fft2(byz)[:(nt+1)//2, :(nx+1)//2])
+    BYZ_4_all_W = np.sum(BYZ, axis=0)
+
+    idx = np.argsort(BYZ_4_all_W)
+    kmodes = idx[-nm:]
+
+    wmodes = np.array([])
+    for i in range(nm):
+        wmodes = np.append(wmodes, np.argmax(BYZ[:,idx[-nm+i]]))
+        #wmodes.append(np.argmax(BYZ[:,idx[-nm+i]]))
+
+    idx = np.argsort(kmodes)
+
+    print(kmodes[idx], wmodes[idx])
+
+    return kmodes[idx], wmodes[idx], BYZ
+
+
+
+def main():
+    # list of modes : m = 1 is for 1 wavelength in the whole domain
+    modes = [4, 8, 16, 32, 64, 128, 256, 512]
+
+    # lists of amplitudes of the magnetic field amplitudes
+    b_amplitudes = [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
+
+    # polarization : -1 for L mode
+    wave_nums, b1 = setOfModes(-1, modes, b_amplitudes, cpp.mpi_rank()+1)
+    simulator = Simulator(gv.sim)
+    simulator.initialize()
+    simulator.run()
+
+    from pybindlibs.cpp import mpi_rank
```

Comment:
rm - use cpp.mpirank()

importing via "pybindlibs" was the old method, the new one is at the top of the file 

```
from pyphare.cpp import cpp_lib
cpp = cpp_lib()
```

this method is so that we can decide which library to use at runtime rather than importing a specific one